packages/node-sdk/paima-utils-backend/src/cde-access.ts

```diff
@@ -167,3 +168,24 @@ export async function getCardanoAddressDelegation(
 ): Promise<string | null> {
   return await internalGetCardanoAddressDelegation(readonlyDBConn, address);
 }
+
+export async function getCardanoAssetUtxosByFingerprint(
+  readonlyDBConn: Pool,
+  address: string,
+  cip14Fingerprint: string
+): Promise<{ txId: string; outputIndex: number; amount: string }[]> {
+  return await internalGetCardanoAssetUtxos(
+    readonlyDBConn,
+    address,
+    'cip14_fingerprint',
+    cip14Fingerprint
+  );
+}
+
+export async function getCardanoAssetUtxosByPolicyId(
+  readonlyDBConn: Pool,
+  address: string,
+  policyId: string
+): Promise<{ txId: string; outputIndex: number; amount: string }[]> {
+  return await internalGetCardanoAssetUtxos(readonlyDBConn, address, 'policy_id', policyId);
+}
```

Comment:
Probably for both of these we need to include the assetName in the return type - otherwise there is no way for the game to know which asset specifically this is referring to

Recall that going from `<policy_id, asset_name>` → fingerprint is one-direction (so unless you also get the policy+asset back, games would have to also include a inverse map which isn't very convenient)

Reply:
Hmmm, yeah. It's actually included in the result... I just forgot to update the type. And also forgot that typescript worked this way.

Currently the result is something like this:

```
utxo {
  txId: '2a35d3e5e55dcf2538826c66989baccfbded226fd70992155f841cd8866b1d6e',
  outputIndex: 0,
  amount: '16502879677629',
  policyId: '919d4c2c9455016289341b1a14dedf697687af31751170d56a31466e',
  assetName: '7449555344'
}
```

Although I guess